mkosi/kmod.py

```diff
@@ -297,30 +255,60 @@ def resolve_module_dependencies(
                 # The file names use dashes, but the module names use underscores. We track the names in
                 # terms of the file names, since the depends use dashes and therefore filenames as well.
                 name = normalize_module_name(value)
-
-                moddep[name] = depends
-                firmwaredep[name] = firmware
-
+                moddep[name] = ModuleDependencyInfo(modules=depends, firmware=firmware)
                 depends = set()
                 firmware = set()
 
+    return moddep
+
+
+@complete_step("Calculating required kernel modules and firmware")
+def resolve_module_dependencies(
+    context: Context,
+    kver: str,
+    modules: Sequence[str],
+) -> tuple[set[Path], set[Path]]:
+    """
+    Returns a tuple of lists containing the paths to the module and firmware dependencies of the given list
+    of module names (including the given module paths themselves). The paths are returned relative to the
+    root directory.
+    """
+    modulesd = Path("usr/lib/modules") / kver
+
+    if (p := context.root / modulesd / "modules.builtin").exists():
+        builtin = set(module_path_to_name(Path(m)) for m in p.read_text().splitlines())
```

Comment:
Since we're touching this anyway.
```suggestion
        builtin = {module_path_to_name(Path(m)) for m in p.read_text().splitlines()}
```